packages/frontend/src/components/voteBox.tsx

```diff
@@ -147,27 +159,67 @@ const VoteBox = ({ isUpvote, closeVote, dataId, isPost }: Props) => {
                             </div>
                         </div>
                     </div>
-                    <div className="epks">
-                        <div
-                            className={epkNonce === -1 ? 'epk chosen' : 'epk'}
-                            onClick={() => setEpkNonce(-1)}
-                        >
-                            Subsidy
+                </div>
+                <div className="white-box">
+                    <div className="info">
+                        <div className="choose-from">
+                            <div className="choices">
+                                <strong
+                                    className={useSubsidy ? 'chosen' : ''}
+                                    onClick={chooseToUseSubsidy}
+                                >
+                                    Rep-Handout
+                                </strong>
+                                <strong
+                                    className={useSubsidy ? '' : 'chosen'}
+                                    onClick={chooseToUsePersona}
+                                >
+                                    Personas
+                                </strong>
+                            </div>
                         </div>
-                        {userContext.currentEpochKeys.map((key, i) => (
-                            <div
-                                className={
-                                    epkNonce === i ? 'epk chosen' : 'epk'
-                                }
-                                key={key}
-                                onClick={() => setEpkNonce(i)}
-                            >
-                                {shortenEpochKey(key)}
+                        {useSubsidy ? (
```

Comment:
Please don't nest ternary operators like this. It is _very_ difficult to reason about this render logic.

Reply:
do you have any idea to simplify the logic? I can only come up with something like this 😅

Reply:
could do it like this

```js
{useSubsidy && userContent.subsidyReputation > 1 && (
    <div className="info-detail">
        <div className="epk chosen">
            <strong>
                {userContext.subsidyReputation}
            </strong>
            <span className="interline"></span>
            {userContext.allEpks[0]}
        </div>
    </div>
)}
{useSubsidy && userContent.subsidyReputation <= 1 && (
    <div className="info-detail">
        You have used all the Rep-Handout ;)
    </div>
)}

{!useSubsidy && userContent.netReputation > 1 && (
    <div className="info-detail">
        <div className="epks">
            {userContext.currentEpochKeys.map(
                (epk, i) => (
                    <div
                        className={
                            i === epkNonce
                                ? 'epk chosen'
                                : 'epk'
                        }
                        onClick={() => setEpkNonce(i)}
                        key={epk}
                    >
                        {shortenEpochKey(epk)}
                    </div>
                )
            )}
        </div>
    </div>
)}
{!useSubsidy && userContent.netReputation <= 1 && (
    <div className="info-detail">
        You don’t have any Rep to use persona yet....
    </div>
)}
```

Reply:
for me it's easier to read because it's like a series of if statements instead of nested if/then, which requires composing the boolean logic up the chain

Reply:
I put them in an individual component, could you check if the logic is good to you? if not, I can change to your kind of logic, it might be clearer